src/gen9_avc_encoder.c

```diff
@@ -794,17 +929,17 @@ gen9_avc_allocate_resources(VADriverContextP ctx,
         goto failed_allocation;
 
     /* scaling related surface   */
-    if (avc_state->mb_status_supported) {
-        i965_free_gpe_resource(&avc_ctx->res_mb_status_buffer);
-        size = (generic_state->frame_width_in_mbs * generic_state->frame_height_in_mbs * 16 * 4 + 1023) & ~0x3ff;
-        allocate_flag = i965_allocate_gpe_resource(i965->intel.bufmgr,
-                                                   &avc_ctx->res_mb_status_buffer,
-                                                   ALIGN(size, 0x1000),
-                                                   "MB statistics output buffer");
-        if (!allocate_flag)
-            goto failed_allocation;
-        i965_zero_gpe_resource(&avc_ctx->res_mb_status_buffer);
-    }
+    // if (avc_state->mb_status_supported) {
+    i965_free_gpe_resource(&avc_ctx->res_mb_status_buffer);
+    size = (generic_state->frame_width_in_mbs * generic_state->frame_height_in_mbs * 16 * 4 + 1023) & ~0x3ff;
+    allocate_flag = i965_allocate_gpe_resource(i965->intel.bufmgr,
+                                               &avc_ctx->res_mb_status_buffer,
+                                               ALIGN(size, 0x1000),
+                                               "MB statistics output buffer");
+    if (!allocate_flag)
+        goto failed_allocation;
+    i965_zero_gpe_resource(&avc_ctx->res_mb_status_buffer);
+    //}
```

Comment:
here we should keep the logic of "// if (avc_state->mb_status_supported) {" and set the "avc_state->mb_status_supported" accordlingly. comment out is a little confused and maybe affect the other platform logic